torch/csrc/jit/api/compilation_unit.cpp

```diff
@@ -0,0 +1,373 @@
+#include <torch/csrc/jit/api/compilation_unit.h>
+#include <torch/csrc/jit/frontend/ir_emitter.cpp>
```

Comment:
Curious about the decision to include `cpp` file. Doing so can invite unwanted side effects.

https://www.cplusplus.com/forum/general/39618/

Reply:
It was very difficult to get the headers right, and I realized that everything Just Worked™ when I included the `cpp` file. I didn't know there were such serious consequences. Thanks for the link!

Does this also apply to tests? I have a `cpp` file in `test_cleanup_passes.cpp`. If I use the header file instead, then the implementation isn't linked by the time the test is compiled. If I put the definition in the header, then there's a problem with `RunCleanupPasses` being defined twice

Reply:
> I have a cpp file in `test_cleanup_passes.cpp`. If I use the header file instead, then the implementation isn't linked by the time the test is compiled.

This suggests that some change needs to be made in some `CMakeLists.txt` to add the new dependency. 🤔

Reply:
Hmm, I'd assume that `run_cleanup_passes.cpp` would need to be specified in `build_variables.bzl::core_sources_full_mobile`, but that doesn't seem to be doing it. 

cc @malfet

Reply:
Resolved by adding `TORCH_API`